src/content/docs/en/guides/deploy/github.mdx

```diff
@@ -138,11 +138,35 @@ export default defineConfig({
     The official Astro [action](https://github.com/withastro/action) scans for a lockfile to detect your preferred package manager (`npm`, `yarn`, `pnpm`, or `bun`). You should commit your package manager's automatically generated `package-lock.json`, `yarn.lock`, `pnpm-lock.yaml`, or `bun.lockb` file to your repository.
     :::
 
-2. On GitHub, go to your repository’s **Settings** tab and find the **Pages** section of the settings.
+2. (Optional) If you pass environment variables to your Astro project during local development or when previewing builds, you will need to define those variables in the `deploy.yml` file in order for them to be processed when you deploy to GitHub Pages.
 
-3. Choose **GitHub Actions** as the **Source** of your site.  
+   ```yaml title="deploy.yml"
+   jobs:
+      build:
+        runs-on: ubuntu-latest
+        steps:
+          - name: Checkout your repository using git
+            uses: actions/checkout@v4
+          - name: Install, build, and upload your site
+            uses: withastro/action@v3
+            env:
+              PUBLIC_EVM_WALLET_ADDRESS: '0x4bFc229A40d41698154336aFF864f61083E76659'
+              SECRET_KEY: 'DO NOT EVER DO THIS IN A PUBLIC REPO'
+   ```
+
+    :::note
+    Unlike values in `.env` files, all strings need to be quote-delimited (as opposed to only strings that contain whitespace).
+    :::
+
+    :::caution
+    If your GitHub repository is public, the environment variables you define in `deploy.yml` will be visible to everyone.  This would be the same as committing your `.env` files.  If you need to provide secrets to your Astro project, please read and understand the [GitHub documentation](https://docs.github.com/en/actions/writing-workflows/choosing-what-your-workflow-does/store-information-in-variables#creating-configuration-variables-for-a-repository) for configuring secrets before continuing.
+    :::
```

Comment:
```suggestion
```
With what we've done above, incorporating these ideas into the instructions, we can get away without these at all!

Reply:
Want to keep this comment alive until the incorrect link in the prior suggestion is updated to this link.